setup.py

```diff
@@ -32,25 +34,20 @@
 from urllib.request import urlretrieve
 
 
-HERE = os.path.dirname(os.path.abspath(__file__))
+HERE = Path(__file__).resolve().parent
 
 DEBUG_COMPILE = "DD_COMPILE_DEBUG" in os.environ
 
 IS_PYSTON = hasattr(sys, "pyston_version_info")
 
-LIBDDWAF_DOWNLOAD_DIR = os.path.join(HERE, os.path.join("ddtrace", "appsec", "_ddwaf", "libddwaf"))
-IAST_DIR = os.path.join(HERE, os.path.join("ddtrace", "appsec", "_iast", "_taint_tracking"))
+LIBDDWAF_DOWNLOAD_DIR = os.path.join(HERE, "ddtrace", "appsec", "_ddwaf", "libddwaf")
+IAST_DIR = os.path.join(HERE, "ddtrace", "appsec", "_iast", "_taint_tracking")
+DDUP_DIR = os.path.join(HERE, "ddtrace", "internal", "datadog", "profiling")
```

Comment:
```suggestion
LIBDDWAF_DOWNLOAD_DIR = HERE / "ddtrace" / "appsec" / "_ddwaf" / "libddwaf"
IAST_DIR = HERE / "ddtrace" / "appsec" / "_iast" / "_taint_tracking"
DDUP_DIR = HERE / "ddtrace" / "internal" / "datadog" / "profiling"
```

Reply:
Ah those might have to be converted to `str` when used perhaps 🤔

Reply:
Converted setup.py to use pathlib when possible in [bc4fc7b](https://github.com/DataDog/dd-trace-py/pull/8203/commits/bc4fc7bb0ab855c6f65529a942528fc0524d7ffc)